datamodel.customer-survey.xml

```diff
@@ -1880,8 +1880,29 @@ EOF
 		{
 			$sBody = $this->Get('email_body');
 		}
-		$sBody .= '<br/><a href="'.$sQuizzUrl.'">'.Dict::S('Survey-notif-linktoquizz').'</a>';
-
+		
+	
+		$aParams = [];
+		$aParams['recipient->object()'] = $oContact;
+		$aParams['quiz_url'] = $sQuizzUrl;
+		
+		$oContactOnBehalfOf = MetaModel::GetObject('Contact', $this->Get('on_behalf_of'), false, true);
+		if(!is_null($oContactOnBehalfOf)) {
+			$aParams['on_behalf_of->object()'] = $oContactOnBehalfOf;
+		}
```

Comment:
Would be more flexible if we could just use `$this$` and all the options from that instead of only the `on_behalf_of` field.

Reply:
Hm, I'm not entirely sure, can I just refer to "this"? (I know there's a function to turn an object into query arguments, just not sure if it would also work out of the box to support `$this->on_behalf_of->attcode$`

Reply:
that's the whole purpose of it indeed..